java-tools/OsmAndServer/src/main/java/net/osmand/server/api/services/EmailRegistryService.java

```diff
@@ -6,18 +6,30 @@
 import java.util.Date;
 import java.util.List;
 
-import org.springframework.beans.factory.annotation.Autowired;
+import net.osmand.server.api.repo.OsmRecipientsRepository;
 import org.springframework.jdbc.core.JdbcTemplate;
 import org.springframework.jdbc.core.RowCallbackHandler;
 import org.springframework.jdbc.core.RowMapper;
 import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
 
 @Service
 public class EmailRegistryService {
-
-	@Autowired
-	private JdbcTemplate jdbcTemplate;
-
+	
+	private final JdbcTemplate jdbcTemplate;
```

Comment:
Doesn't follow conventions we use everywhere  Autowired so here it's not clear why should we do different

Reply:
@Autowired isn't recommended. On the official page Spring, they recommend to use Constructor-based Dependency Injection. @Autowired has many restrictions in use, hides interface and other. Intelij Idea official highlights @Autowired us unrecommended.